### PR TITLE
Appereance tweak

### PR DESCRIPTION
**Pull request by Avassaa**
Fixed feature12

### CODE REVIEW

File: editor-components/feature/feature12/feature12.module.scss [hunk 1]
```diff
@@ -1,52 +1,229 @@
...
+                margin-top: 60px;
+                
+            }
+            
+
+            .card-item-first {
+            
+                width: 100%;
+                align-items: center;
+                background-image: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.75)), url('https://avo.themescamp.com/wp-content/uploads/2020/09/1-2.jpg');
```
Comment (Ayazelar) [inline editor-components/feature/feature12/feature12.module.scss:82]: if this value does not come from the prop the user cannot change the background image

File: editor-components/feature/feature12/feature12.tsx [hunk 2]
```diff
@@ -1,21 +1,65 @@
...
+    this.addProp({
+      type: "string",
+      key:"title-backing-text",
+      displayer:"Title Backing Text",
+      value:"services"
+    })
+    this.addProp({
+      
+        type: "object",
+        key: "message", //TODO:BURASI
```
Comment (Ayazelar) [inline editor-components/feature/feature12/feature12.tsx:39]:
```suggestion
        key: "message",
```

File: editor-components/feature/feature12/feature12.tsx [hunk 3]
```diff
@@ -88,27 +153,78 @@ class Feature12 extends BaseFeature {
...
+                    style={{
+                      width: `${100 / itemCount}%`,
+                      marginBottom: "100px",
+                      height:"50vh"
+                    }}
+                    key={index}
+                  >
+                    {  
+                      <div className={this.decorateCSS("message")}>
+                        <div>
```
Comment (Ayazelar) [inline editor-components/feature/feature12/feature12.tsx:209]: should have className

File: editor-components/feature/feature12/feature12.tsx [hunk 4]
```diff
@@ -88,27 +153,78 @@ class Feature12 extends BaseFeature {
...
+          {firstItem.length > 0 && (
+          <div className={this.decorateCSS("card-item-first")} style={{
+            width: `${100 / itemCount}%`,
+            marginBottom: "100px",
+            height:"50vh"
+          }}>
+            <div className={this.decorateCSS("blurry-background")}></div>
+            <h3 className={this.decorateCSS("title")}>{firstItem[0].value}</h3>
+            <button className={this.decorateCSS("button")} style={{ marginTop: "20px" }}>
+              {firstItem[1].value}
```
Comment (Ayazelar) [inline editor-components/feature/feature12/feature12.tsx:185]: please do not use values like that. check new components

File: editor-components/feature/feature12/feature12.tsx
Comment (Ayazelar) [inline editor-components/feature/feature12/feature12.tsx:1]: We don't recommend using inline CSS as it increases code complexity. It's better to move them to an SCSS file.

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/feature/feature12/feature12.tsx [hunk 5]
```diff
@@ -63,23 +134,37 @@ class Feature12 extends BaseFeature {
...
+
+    const val = this.getPropValue("itemCount") > 5 ? 5 : 4;
     this.addProp({
       type: "number",
       key: "itemCount",
       displayer: "Item count in a row",
-      value: 3,
+      value: val,
     });
   }
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:169]: Consider this format:
```js
 this.addProp({
      type: "number",
      key: "itemCountInRow",
      displayer: "Item Count in a Row",
      value: 4,
      max: 4,
    });
```

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
Please handle the changes.

File: editor-components/feature/feature12/feature12.tsx [hunk 6]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
   render() {
+    const upperTitle = this.getPropValue("upperTitle");
+    const title = this.getPropValue("title");
+    const titleBackingText = this.getPropValue("titleBackingText");
+    const itemCount = this.getPropValue("itemCount");
+    const messageCards = this.castToObject<IMessages[]>("messageCard");
+    const firstItem = this.getPropValue("message");
+    const firstCardTitle = firstItem[0].value;
+    const buttonText = firstItem[1].value;
+    const backgroundImage = firstItem[2].value;
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:184]: don't access values like this. please create a type for firstItem object.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-approach-props)

File: editor-components/feature/feature12/feature12.tsx [hunk 7]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
+   
     return (
-      <div className={this.decorateCSS("container")}>
-        <div className={this.decorateCSS("max-content")}>
+      <div className={this.decorateCSS("container")} style={{ paddingBottom: "0px" }}>
+        <div className={this.decorateCSS("upper-title")} style={{ fontSize: "1.2em", textAlign: "center" }}>
+          {upperTitle}
+        </div>
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:190]: don't use inline css. move them into the scss file.
don't render relevant divs if the prop is empty.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feature/feature12/feature12.tsx [hunk 8]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
+    const messageCards = this.castToObject<IMessages[]>("messageCard");
+    const firstItem = this.getPropValue("message");
+    const firstCardTitle = firstItem[0].value;
+    const buttonText = firstItem[1].value;
+    const backgroundImage = firstItem[2].value;
+   
     return (
-      <div className={this.decorateCSS("container")}>
-        <div className={this.decorateCSS("max-content")}>
+      <div className={this.decorateCSS("container")} style={{ paddingBottom: "0px" }}>
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:187]: don't use inline css. move them into the scss file.

File: editor-components/feature/feature12/feature12.tsx [hunk 9]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
+        <div className={this.decorateCSS("upper-title")} style={{ fontSize: "1.2em", textAlign: "center" }}>
+          {upperTitle}
+        </div>
+        <div className={this.decorateCSS("shadow-header-title")} style={{ fontSize: "12.5em", textAlign: "center", marginBottom: "20px" }}>
+          {titleBackingText}
+        </div>
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:193]: don't use inline css. move them into the scss file.
don't render relevant divs if the prop is empty.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feature/feature12/feature12.tsx [hunk 10]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
+        <div className={this.decorateCSS("header-title")} style={{ fontSize: "5em", textAlign: "center", marginTop: "35px" }}>
+          {title}
+        </div>
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:196]: don't use inline css. move them into the scss file.
don't render relevant divs if the prop is empty.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feature/feature12/feature12.tsx [hunk 11]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
-                <div className={this.decorateCSS("card-item-count")} style={{
-                  width: 90 / this.getPropValue("itemCount") + "%",
-                }}>
-                <div className={this.decorateCSS("message")} key={index}>
-                  <h3 className={this.decorateCSS("title")}>{message.title}</h3>
-                  <p className={this.decorateCSS("long-text")}>{message.description}</p>
-                </div>
-                </div>
-              )
+            {firstItem.length > 0 && (
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:199]: firstItem = prop "message" is an object. it doesn't have length. it is not an array.
it is an object so we should create a type for it.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-approach-props)

File: editor-components/feature/feature12/feature12.tsx [hunk 12]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
+              <div
+                className={this.decorateCSS("card-item-first")}
+                style={{
+                  width: `${100 / itemCount}%`,
+                  backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.75)), url(${backgroundImage})`,
+                 
+                }}
+              >
+                <div className={this.decorateCSS("blurry-background")}></div>
+                <h3 className={this.decorateCSS("title")}>{firstCardTitle}</h3>
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:209]: don't render h3 if firstCardTitle is empty.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feature/feature12/feature12.tsx [hunk 13]
```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
...
+                <button className={this.decorateCSS("button")}>
+                  {buttonText}
+                </button>
```
Comment (dogusmiuw) [inline editor-components/feature/feature12/feature12.tsx:212]: don't render this button if buttonText is empty.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle the changes

**Review by Ayazelar — APPROVED** [review]

**Review by dogusmiuw — APPROVED** [review]

**Pull request merged**